javalin/src/main/java/io/javalin/plugin/CorsPlugin.kt

```diff
@@ -6,24 +6,51 @@ import io.javalin.http.Header.ACCESS_CONTROL_ALLOW_CREDENTIALS
 import io.javalin.http.Header.ACCESS_CONTROL_ALLOW_HEADERS
 import io.javalin.http.Header.ACCESS_CONTROL_ALLOW_METHODS
 import io.javalin.http.Header.ACCESS_CONTROL_ALLOW_ORIGIN
+import io.javalin.http.Header.ACCESS_CONTROL_EXPOSE_HEADERS
 import io.javalin.http.Header.ACCESS_CONTROL_REQUEST_HEADERS
 import io.javalin.http.Header.ACCESS_CONTROL_REQUEST_METHOD
 import io.javalin.http.Header.ORIGIN
 import io.javalin.http.Header.REFERER
 import io.javalin.http.Header.VARY
+import java.util.*
 import java.util.function.Consumer
 
 data class CorsPluginConfig(
     @JvmField var allowCredentials: Boolean = false,
     @JvmField var reflectClientOrigin: Boolean = false,
-    @JvmField var allowedOrigins: Collection<String> = listOf(),
-)
+    private val allowedOrigins: MutableList<String> = mutableListOf(),
+    private val headersToExpose: MutableList<String> = mutableListOf()
+) {
+    fun allowedOrigins(): List<String> = Collections.unmodifiableList(allowedOrigins)
+    fun headersToExpose(): List<String> = Collections.unmodifiableList(headersToExpose)
+
+    fun anyHost() {
+        allowedOrigins.add("*")
+    }
+
+    fun allowHost(origin: String) {
+        allowedOrigins.add(origin.removeSuffix("/"))
+    }
+
+    // TODO: unsure if that is the desired api. Maybe just accept a collection instead?
+    fun allowHost(origin: String, vararg others: String) {
```

Comment:
Why is the single String method necessary?

Reply:
Now that you have said it. Yes we don't need it. 😅